Blocks/HTTP/test.cc

```diff
@@ -427,6 +436,10 @@ TEST(HTTPAPI, PostCerealizableObject) {
   });
   EXPECT_EQ("Data: {\"data\":{\"x\":42,\"s\":\"foo\"}}",
             HTTP(POST(Printf("http://localhost:%d/post", FLAGS_net_api_test_port), SerializableObject())).body);
+
+  EXPECT_EQ(
+      "Data: {\"custom_name\":{\"y\":100}}",
```

Comment:
Ah, no, you can always do 'response(my_object, "name")', or 'JSON(my_object,
"custom_name")', or a couple other things. This change is to merely make it
easier to change the default "data".

On Wednesday, July 22, 2015, John Babak <notifications@github.com> wrote:

> In Blocks/HTTP/test.cc
> <https://github.com/c5t/Current/pull/225#discussion_r35267119>:
>
> > @@ -427,6 +436,10 @@ TEST(HTTPAPI, PostCerealizableObject) {
> >    });
> >    EXPECT_EQ("Data: {\"data\":{\"x\":42,\"s\":\"foo\"}}",
> >              HTTP(POST(Printf("http://localhost:%d/post", FLAGS_net_api_test_port), SerializableObject())).body);
> > +
> > +  EXPECT_EQ(
> > +      "Data: {\"custom_name\":{\"y\":100}}",
>
> I see this is by design. I think it's not a good design to always force a
> specific name for an instance of some type. This way the type becomes
> non-reusable. Imagine I've got a Slice and it has a JSONEntryName that
> returns "slice". If I wanted two API endpoints return different kinds of
> slices, I would make one endpoint name the response instance a
> "first_endpoint_slice", the other "second_endpoint_slice", both return
> instances of Slice. With the new architecture it looks like a wrong way
> to go, so I should make a new (maybe derived) type to name my responses
> differently.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/c5t/Current/pull/225/files#r35267119>.
>


-- 
And, as always, have a nice day!
Dima

http://dimakorolev.com/ | http://www.linkedin.com/in/dimakorolev |
dmitry.korolev@gmail.com | +1 (312) 593 2783